lib/puppet/parser/functions/ip_for_network.rb

```diff
@@ -12,7 +12,7 @@ module Puppet::Parser::Functions
 
     range = IPAddr.new(args[0])
     facts = compiler.node.facts.values
-    ip_addresses = facts.select { |key, value| key.match /^ipaddress/ }
+    ip_addresses = facts.select { |key, value| key =~ /^ipaddress/ }
```

Comment:
This is to support the lower version of puppet. I was using puppet-3.1.1-1.el6 which results in failure due to non-support of key.match fucntion.